packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -302,20 +302,105 @@ List<Plugin> findPlugins(FlutterProject project) {
   }
   packages.forEach((String name, Uri uri) {
     final Uri packageRoot = uri.resolve('..');
-    final Plugin plugin = _pluginFromPubspec(name, packageRoot);
+    final Plugin plugin = _pluginFromPackage(name, packageRoot);
     if (plugin != null) {
       plugins.add(plugin);
     }
   });
   return plugins;
 }
 
+void _addPluginsToPlatform(List<Plugin> plugins, FlutterProjectPlatform platform, Map<String, dynamic> pluginsMap) {
+  if (!platform.existsSync()) {
```

Comment:
I got rid of this and added to the map directly. I was trying to avoid writing platforms that have no plugins but as you say, there's no much point to that. If there are no plugins at all, the file won't be written.